tests/test_benchmark.py

```diff
@@ -5,30 +5,136 @@
-        diff += periodicBoxSize[0] * np.floor((atom[0] - diff[0]) / periodicBoxSize[0][0])
-        new_coords.append(atom - diff)
+    harmonic_bond_potential = unbound_potentials[0]
+    bond_list = get_bond_list(harmonic_bond_potential)
+    if hmr:
+        dt = 2.5e-3
+        masses = apply_hmr(masses, bond_list)
+    if seed is None:
+        seed = np.random.randint(np.iinfo(np.int32).max)
+    intg = LangevinIntegrator(temperature, dt, 1.0, np.array(masses), seed).impl()
+
+    bps = []
+
+    for potential, params in zip(unbound_potentials, sys_params):
+        bps.append(potential.bind(params).bound_impl(precision=np.float32))  # get the bound implementation
+
+    baro_impl = None
+    if barostat_interval > 0:
+        group_idxs = get_group_indices(bond_list)
+        baro = MonteCarloBarostat(
+            x0.shape[0],
+            pressure,
+            temperature,
+            group_idxs,
+            barostat_interval,
+            seed,
+        )
+        baro_impl = baro.impl(bps)
 
-    return np.array(new_coords)
+    ctxt = custom_ops.Context(
+        x0,
+        np.zeros_like(x0),
+        host_box,
+        intg,
+        bps,
+        barostat=baro_impl,
+    )
+    steps = n_frames * frame_interval
+    _, coords, boxes = ctxt.multiple_steps(np.ones(steps) * lamb, 0, frame_interval)
+    assert coords.shape[0] == n_frames, f"Got {coords.shape[0]} frames, expected {n_frames}"
+    return unbound_potentials, sys_params, coords, boxes, ligand_idxs
+
+
+def benchmark_potential(
+    label,
+    potential,
+    precision,
+    params,
+    coords,
+    boxes,
+    lambdas,
+    verbose=True,
+    num_batches=5,
+    compute_du_dx=True,
+    compute_du_dp=True,
+    compute_du_dl=True,
+    compute_u=True,
+):
+    if precision == np.float32:
+        label = label + "_f32"
+    else:
+        label = label + "_f64"
+    unbound = potential.unbound_impl(precision=precision)
+    start = time.time()
+    batch_times = []
+    frames = coords.shape[0]
+    param_batches = params.shape[0]
+    num_lambs = len(lambdas)
+    runs_per_batch = frames * param_batches * num_lambs
+    for _ in range(num_batches):
+        batch_start = time.time()
+        _, _, _, _ = unbound.execute_selective_batch(
+            coords,
+            params,
+            boxes,
+            lambdas,
+            compute_du_dx,
+            compute_du_dp,
+            compute_du_dl,
+            compute_u,
+        )
+        batch_end = time.time()
+        delta = batch_end - batch_start
+
+        batch_times.append(delta)
+        runs_per_second = runs_per_batch / np.mean(batch_times)
+
+        if verbose:
+            print(f"executions per second: {runs_per_second:.3f}")
+    print(
+        f"{label}: N={coords.shape[1]} Frames={frames} Params={param_batches} Lambdas={num_lambs} speed: {runs_per_second:.2f} executions/seconds (ran {runs_per_batch * num_batches} potentials in {(time.time() - start):.2f}s)"
```

Comment:
Nit: preference between `sum(batch_times)` and `time.time() - start`? (to exclude some setup and printing)

Reply:
I prefer the latter as it is consistent with the other benchmarking. We may want to revisit how we do benchmarking in the next deboggling week